src/Uno.UI.Composition/Composition/BorderVisual.skia.cs

```diff
@@ -81,8 +84,11 @@ private protected override void OnPropertyChangedCore(string? propertyName, bool
 				_borderPathValid = false; // to update _borderPath if previously skipped
 				if (BorderBrush is not null && _borderShape is null)
 				{
-					// we need this to track get notified on brush updates.
-					SetProperty(ref _borderShape, Compositor.CreateSpriteShape());
+					// we need this to get notified on brush updates
+					// (BorderBrush internals change -> BorderShape is notified through FillBrush -> render invalidation)
+					BorderShape = Compositor.CreateSpriteShape();
+
+					_borderShape!.Geometry = Compositor.CreatePathGeometry();
```

Comment:
This will probably led to 2 invalidation, it's usually better to fully configure the object before assigning it rather than assign it and then update its properties.